routing/dht/dht_net.go

```diff
@@ -227,3 +228,23 @@ func (ms *messageSender) SendRequest(ctx context.Context, pmes *pb.Message) (*pb
 
 	return mes, nil
 }
+
+func (ms *messageSender) ctxReadMsg(ctx context.Context, mes *pb.Message) error {
+	t := time.NewTimer(time.Second * 30)
```

Comment:
theres a bug in utp that surfaces occasionally that will cause it to hang. Theres nothing we can do to avoid this other than fix utp, or not use utp. The hang that i wrote this PR to 'solve' was caused by this utp bug.